packages/xcoobee-cookie-kit-react/src/CookieKitContainer.js

```diff
@@ -159,66 +167,70 @@ export default class CookieKitContainer extends React.PureComponent {
     }
   }
 
-  componentDidMount() {
-    if (this.state.countryCode) {
-      this.getCookieConsents();
-    } else {
-      CookieConsentsManager.fetchCountryCode()
+  onLoginStatusChange(loginStatus) {
+    // console.log("CookieKitContainer#onLoginStatusChange");
+    const { isLoginStatusChecked } = this.state;
+
+    this.setState({ loginStatus });
+
+    if (!isLoginStatusChecked) {
+      this.setState({ isLoginStatusChecked: true });
+      this.getCountryCode()
         .catch((error) => {
-          // console.log("CookieKitContainer#componentDidMount#fetchCountryCode#catch");
           console.error(error);
           return null;
         })
-        .then((countryCode) => {
-          saveCountryCode(countryCode);
-          this.setState({ countryCode });
+        .then((cCode) => {
+          saveCountryCode(cCode);
```

Comment:
I think saving of the country code and updating the `countryCode` state should be done in `getCountryCode`. That way, we are not saving and updating the state in the case we already know the country code. (Calling `this.setState` will cause a re-render even if ultimately the value did not change.) That is, I think only calling `this.getCookieConsents` belongs in this `then` callback.